### PR TITLE
vtpm fix : use domain uuid instead of full domain name

### PR DESCRIPTION
**Pull request by shjala**
Domain name contains UUID, version and app number,the version number might get changed, so use only the
UUID part to preserve the vtpm state.

---

This PR is based on a observation in @OhmSpectator code comment : 
> each time the configuration is changed, it updates the domain name, increasing the counter after UUID

I'll be happy if someone point me to part of the code that actually makes this change to ~app~ version number, I couldn't find it myself.

### CODE REVIEW

**Review by OhmSpectator — APPROVED** [review]
Lucky you, the tests are fine... I'm fixing kvm_tets for an hour now =D

**Comment by OhmSpectator** [comment]
> I'll be happy if someone point me to part of the code that actually makes this change to app number, I couldn't find it myself.

Config Num comes from the cloud. They increase it with each config update.

**Comment by OhmSpectator** [comment]
The virtualization test suite has failed three times. Strange.

**Comment by OhmSpectator** [comment]
Yeah, it looks like something is broken...

```
[stdout]
Docker app's state test
=== RUN   TestAppStatus
apps: '[eclient]' state: 'RUNNING' secs: 1800
time: 2024-09-19T02:04:00.913252133Z out: 	appName eclient state changed to UNKNOWN
time: 2024-09-19T02:04:03.570037979Z out: 	appName eclient state changed to RESOLVING_TAG
time: 2024-09-19T02:04:04.570522093Z out: 	appName eclient state changed to DOWNLOAD_STARTED
time: 2024-09-19T02:04:05.571844887Z out: 	appName eclient state changed to DOWNLOAD_STARTED (0%)
time: 2024-09-19T02:04:08.574364348Z out: 	appName eclient state changed to DOWNLOAD_STARTED (11%)
time: 2024-09-19T02:04:08.574380459Z out: 	appName eclient state changed to DOWNLOAD_STARTED (23%)
time: 2024-09-19T02:04:09.57494185Z out: 	appName eclient state changed to DOWNLOAD_STARTED (64%)
time: 2024-09-19T02:04:09.574978076Z out: 	appName eclient state changed to LOADING
time: 2024-09-19T02:04:10.576055852Z out: 	appName eclient state changed to CREATING_VOLUME
time: 2024-09-19T02:04:14.578651073Z out: 	appName eclient state changed to INSTALLED
time: 2024-09-19T02:04:15.579136424Z out: 	appName eclient state changed to BOOTING
time: 2024-09-19T02:05:54.653641582Z out: 	appName eclient state changed to HALTING: [description:"Giving up waiting to connect to QEMU Monitor Protocol socket /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp from VM 7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1, error: [attempt 1] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: ''); [attempt 2] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: ''); [attempt 3] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: '')" timestamp:{seconds:1726711530 nanos:262492703} severity:SEVERITY_ERROR]
time: 2024-09-19T02:15:47.104054215Z out: 	appName eclient state changed to BOOTING
time: 2024-09-19T02:17:26.189625019Z out: 	appName eclient state changed to HALTING: [description:"Giving up waiting to connect to QEMU Monitor Protocol socket /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp from VM 7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1, error: [attempt 1] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: ''); [attempt 2] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: ''); [attempt 3] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: '')" timestamp:{seconds:1726712221 nanos:781452343} severity:SEVERITY_ERROR]
time: 2024-09-19T02:27:09.631665355Z out: 	appName eclient state changed to BOOTING
time: 2024-09-19T02:28:48.707869707Z out: 	appName eclient state changed to HALTING: [description:"Giving up waiting to connect to QEMU Monitor Protocol socket /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp from VM 7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1, error: [attempt 1] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp': err: 'dial unix /run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1/qmp: connect: connection refused'; (JSON response: ''); [attempt 2] qmp status failed for QMP socket '/run/hypervisor/kvm/7418b0e3-96e3-4e0f-b977-3720d7464f1c.1.1
```

**Comment by eriknordmark** [comment]
> > I'll be happy if someone point me to part of the code that actually makes this change to app number, I couldn't find it myself.
> 
> Config Num comes from the cloud. They increase it with each config update.

It is the version number (and not the app number) which is changed, right? If so it makes sense to update the PR description.

File: pkg/pillar/hypervisor/kvm.go [hunk 1]
```diff
@@ -1181,23 +1181,35 @@ func getQmpListenerSocket(domainName string) string {
 func (ctx KvmContext) VirtualTPMSetup(domainName, agentName string, ps *pubsub.PubSub, warnTime, errTime time.Duration) error {
 	if ps != nil {
 		wk := utils.NewWatchdogKick(ps, agentName, warnTime, errTime)
-		return requestVtpmLaunch(domainName, wk, swtpmTimeout)
+		domainUUID, _, _, err := types.DomainnameToUUID(domainName)
+		if err != nil {
+			return fmt.Errorf("failed to extract UUID from domain name: %v", err)
+		}
+		return requestVtpmLaunch(domainUUID.String(), wk, swtpmTimeout)
```
Comment (eriknordmark) [inline pkg/pillar/hypervisor/kvm.go:1188]: Nit, but if the argument type was uuid.UUID and not a string then the compiler would help check.

**Comment by OhmSpectator** [comment]
@shjala, will you take a look at why the same Eden test fails all the time?

**Comment by shjala** [comment]
> @shjala, will you take a look at why the same Eden test fails all the time?

All are green except virtualization, do you mean that or there is more?

**Comment by OhmSpectator** [comment]
Yeah, I mean that one. I don't see the same test failing in other PRs all the time. Maybe it's just a coincidence, but it's better to check.

**Comment by shjala** [comment]
@OhmSpectator should be fixed now, rerun?

**Comment by OhmSpectator** [comment]
All green! What exactly have you done? Just rebase?

File: pkg/pillar/hypervisor/kvm.go [hunk 2]
```diff
@@ -799,7 +799,12 @@ func (ctx KvmContext) Setup(status types.DomainStatus, config types.DomainConfig
 	// check if vTPM is enabled
 	swtpmCtrlSock := ""
 	if status.VirtualTPM {
-		swtpmCtrlSock = fmt.Sprintf(types.SwtpmCtrlSocketPath, domainName)
+		domainUUID, _, _, err := types.DomainnameToUUID(domainName)
```
Comment (shjala) [inline pkg/pillar/hypervisor/kvm.go:802]: @OhmSpectator this is the part I missed to update in the previous version, prevent VM from running becuse SWTPM socket is invalid.

**Comment by shjala** [comment]
> All green! What exactly have you done? Just rebase?

Eden rerun is appreciated.

**Review by OhmSpectator — APPROVED** [review]

**Pull request merged**